### DISCUSSION TITLE
OneDrive Client Version v2.4.24 | 2.4.25 - Deprecation of sha1Hash on OneDrive Personal

### DISCUSSION THREAD

**Discussion by abraunegg** [abraunegg/onedrive#2441] [General]
Microsoft has announced the deprecation of sha1Hash on OneDrive Personal starting July 2023.

OneDrive Client Version v2.4.25 has been released to ensure that there are no impacts because of this announcement, however it is *critical* that you update to v2.4.25 as soon as possible to ensure you are not impacted by the older (v2.4.23 and below) which has to the potential to use sha1 hashes.

v2.4.25 also fixes a previously unknown issue of zero byte files having no hash to compare - thus, with the change required for the depreciation, this actually surfaced a long standing API issue which has now been worked around.

When running this application version, any file that *does* not have a valid hash, will present the following warning:
```
WARNING: OneDrive API inconsistency - this file does not have any hash: XXXX
```
This warning is *not* a bug or issue. It is a OneDrive API consistency where your file, stored on OneDrive does not contain the required hash types that this application is looking for as part of the depreciation of the sha1hash by Microsoft. 

At the moment , if you get this sort of message:
```
WARNING: OneDrive API inconsistency - this file does not have any hash: EB356FD8AA88F1C3!129239
```
or
```
WARNING: OneDrive API inconsistency - this file does not have any hash: 01S4CBCJG6N3O77JM6VZGJH3MEBP5D2T6Q
```
This is indicative of the item, that is missing the hash is either:
* A Microsoft OneNote items stored on OneDrive
* Items that exist in your online 'Recycle Bin'
* The parent item has been excluded by your application configuration
* Items that exist in OneDrive Personal Vaults

These sort of warning messages can be safely ignored. Please do not raise a question or issue ticket. This will be further cleaned up in v2.5.0 so these messages are not causing alarm.

If you get this sort of message, that contains a file path:
```
WARNING: OneDrive API inconsistency - this file does not have any hash: Folder/subfolder/subfolder_two/document.extention
```
This is indicative of the item, as stored online by the OneDrive platform is missing the required hash. This is a OneDrive API issue and needs to be reported to the OneDrive API team - post your error and details here: https://github.com/OneDrive/onedrive-api-docs/issues/1704 - this is a Microsoft issue that needs to be resolved by Microsoft.

It will take a few days for package maintainers to update their packages - so please keep an eye out for new packages for your distribution, otherwise you will need to build from source.

**References:**
* https://devblogs.microsoft.com/microsoft365dev/deprecation-of-sha1hash-on-onedrive-personal/

**Comment by kayzzen01** [comment]
Hi,

I noticed this hash warning, even downloaded and uploaded all my files. Checking the notices I noticed that they were from OneNote. Glad I don't have to worry!

**Reply by 1040170566** [reply]
Yes, I met this warning because of OneNote link. Do you have any solutions for it?

**Reply by abraunegg** [reply]
Solutions - as per above:
1. Ignore it
2. Upgrade your client to v2.5.0 when available
